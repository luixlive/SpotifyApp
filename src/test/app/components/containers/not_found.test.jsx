import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import {
  ConnectedNotFound,
  NotFound,
} from './../../../../app/components/containers/not_found';
import initialState from './../../../test_utils/initial_state';
import injectRouter from './../../../test_utils/inject_router';
import {
  loginContainer as componentProps,
} from './../../../test_utils/components_props';

describe('App Components - NotFound', () => {
  describe('Snapshots', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders no mobile', () => {
      const rendered =
        renderer.create(injectRouter(() => <NotFound {...props} />)).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders mobile', () => {
      props.isDeviceMobile = true;
      const rendered =
        renderer.create(injectRouter(() => <NotFound {...props} />)).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Provider', () => {
    const mockStore = configureStore();
    let store;
    let wrapper;
    beforeEach(() => {
      store = mockStore(initialState);
      wrapper = mount(injectRouter(() =>
        <Provider store={store}><ConnectedNotFound /></Provider>));
    });

    it('renders', () => {
      expect(wrapper.find(ConnectedNotFound).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(NotFound).props()))
        .toEqual(Object.keys(componentProps));
    });
  });
});
