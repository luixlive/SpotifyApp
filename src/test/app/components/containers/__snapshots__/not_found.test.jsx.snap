// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - NotFound Snapshots renders mobile 1`] = `
<div
  className="ui center aligned container"
>
  <h1
    className="ui header"
    style={
      Object {
        "fontSize": "2em",
        "marginTop": "1em",
      }
    }
  >
    Error 404
  </h1>
  <h3
    className="ui header"
  >
    Page not found, go back to 
    <a
      href="/stats"
      onClick={[Function]}
    >
      home
    </a>
     page.
  </h3>
</div>
`;

exports[`App Components - NotFound Snapshots renders no mobile 1`] = `
<div
  className="ui center aligned container"
>
  <h1
    className="ui header"
    style={
      Object {
        "fontSize": "4em",
        "marginTop": "1.5em",
      }
    }
  >
    Error 404
  </h1>
  <h2
    className="ui header"
  >
    Page not found, go back to 
    <a
      href="/stats"
      onClick={[Function]}
    >
      home
    </a>
     page.
  </h2>
</div>
`;
